File: 2_module/Prim/Prim.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n, m;
    cin >> n >> m;
    vector <vector <pair <int, int> > > g (n);
    int x, y, dist;
    for (int i = 0; i < m; i++){
        cin >> x >> y >> dist;
        g[x].push_back({y, dist});
        g[y].push_back({x, dist});
    }
    int used[n] = {};
    int used_v = 1, ans = 0;
    used[0] = 1;
    while(used_v < n){
        int mun = 1e9, vmun;
        for (int i = 0; i < n; i++){
            if (used[i]){
                for (auto j : g[i]){
                    if (!used[j.first]){
                        if (j.second < mun){
                            mun = j.second;
                            vmun = j.first;
                        }
                    }
                }
            }
        }
        used[vmun] = 1;
        ans += mun;
        used_v++;
    }
    cout << ans;
}

File: 2_module/Prim/Prim_go/Prim.go
package main

import "fmt"

type pair struct {
	first, second int
}

func main() {
	var n, m int
	fmt.Scan(&n, &m)
	g := make([][]pair, n, n)
	for i := range g {
		g[i] = make([]pair, 0, 0)
	}
	var x, y, dist int
	for i := 0; i < m; i++ {
		fmt.Scan(&x, &y, &dist)
		g[x] = append(g[x], pair{y, dist})
		g[y] = append(g[y], pair{x, dist})
	}
	used := make([]bool, n, n)
	used_v := 1
	ans := 0
	used[0] = true
	for ; used_v < n; used_v++ {
		mun := 10000000
		vmun := 0
		for i := 0; i < n; i++ {
			if used[i] {
				for j := range g[i] {
					if !used[g[i][j].first] {
						if g[i][j].second < mun {
							mun = g[i][j].second
							vmun = g[i][j].first
						}
					}
				}
			}
		}
		used[vmun] = true
		ans += mun
	}
	fmt.Println(ans)
}
